src/core/shape_inference/include/range_shape_inference.hpp

```diff
@@ -19,10 +19,6 @@ std::vector<TRShape> range_shape_infer(const Node* op,
                                        const ITensorAccessor& tensor_accessor) {
     NODE_VALIDATION_CHECK(op, (input_shapes.size() == 3));
 
-    NODE_VALIDATION_CHECK(op, input_shapes[0].rank().compatible(0), "'start' input is not a scalar");
-    NODE_VALIDATION_CHECK(op, input_shapes[1].rank().compatible(0), "'stop' input is not a scalar");
-    NODE_VALIDATION_CHECK(op, input_shapes[2].rank().compatible(0), "'step' input is not a scalar");
```

Comment:
Removal of those checks, makes the shape of related `start/stop/step` inputs not validated at all.
It means any shape will be allowed, what is not correct. The check for scalar can be optionally extended to allow the input shape to be 0D or 1D with single element.
Because 1D with more elements than one, is not correct as well, so it should be ensured.